node/src/protocol.rs

```diff
@@ -67,6 +67,24 @@ impl Payload for Message {
             Message::FinalitySignature(_) => MessageKind::Consensus,
         }
     }
+
+    #[inline]
+    fn incoming_resource_estimate(&self) -> u32 {
+        match self {
+            Message::Consensus(_) => 0,
+            Message::DeployGossiper(_) => 0,
+            Message::AddressGossiper(_) => 0,
+            Message::GetRequest { tag, .. } | Message::GetResponse { tag, .. } => match tag {
+                Tag::Deploy => 1,
+                Tag::Block => 0,
+                Tag::GossipedAddress => 0,
+                Tag::BlockByHeight => 0,
+                Tag::BlockHeaderByHash => 0,
+                Tag::BlockHeaderAndFinalitySignaturesByHeight => 0,
```

Comment:
By setting `0` for all the `Tag::Block*` variants, we're allowing for almost unlimited joining, even for non-validating nodes. Is that intentional?

Reply:
Yes, we're being very conservative with the introduction of this feature. We can tweak the numbers if we have to though.

The reason deploys are throttled as the highest priority is because they offer the largest possible size. AFAIK any other message is limited to a few K in size, while deploys can be 1M.

Reply:
Block's limit is 10M.

Reply:
Ah, but block's limit takes into account its constituents deploys so a block _message_ should be smaller than 10MB :+1: